blas/impl/KokkosBlas2_ger_impl.hpp

```diff
@@ -0,0 +1,339 @@
+        });
+      }
+    }
+  }
+
+  // LayoutRight version: one team per row
+  KOKKOS_INLINE_FUNCTION void operator()( TwoLevelGER_LayoutRightTag
+                                        , const member_type & team
+                                        ) const {
+    using KAT = Kokkos::ArithTraits<typename AViewType::non_const_value_type>;
+
+    if (alpha_ == KAT::zero()) {
+      // Nothing to do
+    }
+    else {
+      const IndexType    N      ( A_.extent(1) );
+      const IndexType    i      ( team.league_rank() );
+      const A_value_type x_fixed( x_(i) );
+      if (justTranspose_) {
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, N), [&](const IndexType & j) {
+          A_(i,j) += A_value_type( alpha_ * x_fixed * y_(j) );
+        });
+      }
+      else {
+        Kokkos::parallel_for(Kokkos::TeamThreadRange(team, N), [&](const IndexType & j) {
+          A_(i,j) += A_value_type( alpha_ * x_fixed * KAT::conj( y_(j) ) );
+        });
+      }
+    }
+    team.team_barrier();
+  }
+
+private:
+  bool                           justTranspose_;
+  AlphaCoeffType                 alpha_;
+  typename XViewType::const_type x_;
+  typename YViewType::const_type y_;
+  AViewType                      A_;
+};
+
+// Two-level parallel version of GER.
+template <class XViewType, class YViewType, class AViewType,
+          class IndexType = typename AViewType::size_type>
+void twoLevelGer( const typename AViewType::execution_space  & space
+                , const          char                          trans[]
+                , const typename AViewType::const_value_type & alpha
+                , const          XViewType                   & x
+                , const          YViewType                   & y
+                , const          AViewType                   & A
+                ) {
+  KOKKOS_IMPL_DO_NOT_USE_PRINTF( "Entering IMPL twoLevelGer(), AViewType = %s\n", typeid(AViewType).name() );
+  static_assert(Kokkos::is_view<XViewType>::value, "XViewType must be a Kokkos::View.");
+  static_assert(Kokkos::is_view<YViewType>::value, "YViewType must be a Kokkos::View.");
+  static_assert(Kokkos::is_view<AViewType>::value, "AViewType must be a Kokkos::View.");
+
+  static_assert(static_cast<int>(XViewType::rank) == 1, "XViewType must have rank 1.");
+  static_assert(static_cast<int>(YViewType::rank) == 1, "YViewType must have rank 1.");
+  static_assert(static_cast<int>(AViewType::rank) == 2, "AViewType must have rank 2.");
+
+  static_assert(std::is_integral<IndexType>::value, "IndexType must be an integer");
+
+  using KAT = Kokkos::ArithTraits<typename AViewType::non_const_value_type>;
+
+  if (y.extent(0) == 0) {
+    // no entries to update
+    return;
+  }
+  else if (x.extent(0) == 0) {
+    // no entries to update
+    return;
+  }
+  else if (alpha == KAT::zero()) {
+    // no entries to update
+    return;
+  }
+
+  using execution_space = typename AViewType::execution_space;
```

Comment:
Should be removed after adding the new template parameter

Reply:
Done.